### ISSUE TITLE
SQL Server Bug v2.6.1.beta and v.2.5.0 - Default values overwritten to NULL

### ISSUE THREAD

**Issue by leoalves** [hasura/graphql-engine#8443]
### Version Information
Server Version: v2.5.0 and v.2.6.beta

### What is the expected behaviour?
Hasura is generating queries with NULL values, instead of omitting the column.  This overwrites default values set into the fields or generates constraint validation errors for non-null fields.

### How to reproduce the issue?
1.  connect SQL server database to Hasura
2. create table with default value
```SQL 
CREATE TABLE defaultValueTest
(
    id         uniqueidentifier PRIMARY KEY DEFAULT NEWID(),
    name       VARCHAR(255) NOT NULL,    
    created_at DATETIME2 NOT NULL DEFAULT SYSDATETIME()
);
```
3. Track table
4. Insert record into the table passing only the `name` field
```
mutation {
  insert_defaultValueTest_one(object: {
    name: "a"
  }) {
    id
  }
}
```
5. Hasura will return an error saying the id field is mandatory and cannot be `NULL`

### Can you identify the location in the source code where the problem exists?
I believe this merged pull request is responsible for this issue
https://github.com/hasura/graphql-engine/commit/92ab93775f5d79aa9407999acb0f85c948d3155d#diff-d79e7a125fc4aff3fef93e498097bd00b33e8eee54a98be0d8bd20c2ddfef52bL28-L29

If you compare the `VALUES` from the generated query you can see that in the previous version only one value is passed.  In the new one all other fields are set to NULL.

**Comment by sassela** [comment]
@leoalves thanks for the bug report! We'll triage this and get back to you with any questions or updates soon.

**Comment by sassela** [comment]
We've flagged this as an urgent bug, but we suspect there's a fix already on the main branch awaiting release. We'll regression test it, release a patch and keep this issue updated once it's available.

Until then, here's the [as-yet-unreleased change we think resolves it](https://github.com/hasura/graphql-engine/commit/df36cdac09cd82fff5b3d441bb40f1f51125c3f7), which you can access early if needed:
🐳 Docker image: hasurabuild/graphql-engine:mono-branch-main-91f80902a
📦 Heroku app: https://mono-branch-main-91f80902a.herokuapp.com/

**Comment by AndrewCraswell** [comment]
I've just hit the same bug using Hasura Cloud. Unfortunately this is a big blocker :/

**Comment by sassela** [comment]
This will be fixed in [release v2.6.1](https://github.com/hasura/graphql-engine/releases) going out today. I'll update here once it's available.

**Comment by sassela** [comment]
hey folks, thank you again for your patience. Confirming the fix for this bug is now available on [v2.6.1](https://github.com/hasura/graphql-engine/releases/tag/v2.6.1). Please comment here or reopen the issue if any problems.
